modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/viewmodel/ShelfBottomSheetViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package au.com.shiftyjelly.pocketcasts.player.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.models.to.Transcript
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.TranscriptsManager
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+@HiltViewModel(assistedFactory = ShelfBottomSheetViewModel.Factory::class)
+class ShelfBottomSheetViewModel @AssistedInject constructor(
+    @Assisted private val episodeId: String?,
+    private val transcriptsManager: TranscriptsManager,
+) : ViewModel() {
+    private var _uiState: MutableStateFlow<UiState> = MutableStateFlow(UiState())
+    val uiState: StateFlow<UiState> = _uiState
+
+    init {
+        viewModelScope.launch {
+            episodeId?.let {
+                transcriptsManager.observerTranscriptForEpisode(episodeId)
+                    .distinctUntilChanged { t1, t2 -> t1?.episodeUuid == t2?.episodeUuid }
```

Comment:
This can be alternatively expressed as `distinctUntilChangedBy { it?.episodeUuid }`. No need to change tough if you prefer this version.

Reply:
I like it. Applied in 353387b.